designer/client/__tests__/link-create.jest.tsx

```diff
@@ -130,7 +145,95 @@ test("links are correctly generated when the form is submitted", () => {
   fireEvent.click(getByRole("button"));
   expect(save).toBeCalledTimes(2);
 
-  expect(save.mock.calls[1][0].pages[1].next).toContainEqual({
+  expect(save.mock.calls[1][0].pages[2].next).toContainEqual({
+    path: "/first-page",
+  });
+});
+
+test("links are correctly generated when the form is submitted", () => {
+  const data = {
+    ...rawData,
+    conditions: [
+      {
+        name: "hasUKPassport",
+        displayName: "hasUKPassport",
+        value: {
+          name: "hasUKPassport",
+          conditions: [
+            {
+              field: {
+                name: "ukPassport",
+                type: "YesNoField",
+                display: "Do you have a UK passport?",
+              },
+              operator: "is",
+              value: {
+                type: "Value",
+                value: "yes",
+                display: "Yes, I have a UK passport",
+              },
+            },
+          ],
+        },
+      },
+      {
+        name: "doesntHaveUKPassport",
+        displayName: "doesntHaveUKPassport",
+        value: {
+          name: "doesntHaveUKPassport",
+          conditions: [
+            {
+              field: {
+                name: "ukPassport",
+                type: "YesNoField",
+                display: "Do you have a UK passport?",
+              },
+              operator: "is",
+              value: {
+                type: "Value",
+                value: "no",
+                display: "No, I do not have a UK passport",
+              },
+            },
+          ],
+        },
+      },
+    ],
+  };
+  const save = jest.fn();
+  const { getByTestId, getByRole } = customRender(<LinkCreate />, {
+    data,
+    save,
+  });
+  fireEvent.change(getByTestId("link-source"), {
+    target: { value: "/first-page" },
+  });
+  fireEvent.change(getByTestId("link-target"), {
+    target: { value: "/summary" },
+  });
+  fireEvent.change(getByTestId("select-condition"), {
+    target: { value: "hasUKPassport" },
+  });
+  fireEvent.click(getByRole("button"));
+  expect(save).toBeCalledTimes(1);
+  expect(save.mock.calls[0][0].pages[0].next).toContainEqual({
+    path: "/summary",
+    condition: "hasUKPassport",
+  });
+
+  fireEvent.change(getByTestId("link-source"), {
+    target: { value: "/summary" },
+  });
+  fireEvent.change(getByTestId("link-target"), {
+    target: { value: "/first-page" },
+  });
+  fireEvent.change(getByTestId("select-condition"), {
+    target: { value: "hasUKPassport" },
+  });
+  fireEvent.click(getByRole("button"));
+  expect(save).toBeCalledTimes(2);
+
+  expect(save.mock.calls[1][0].pages[2].next).toContainEqual({
     path: "/first-page",
```

Comment:
Might need some clarification on how mocking works in this project, does this make a call to a controller?

Reply:
have a look at customRender 

```
export const customRender = (children, providerProps = dataValue) => {
  return render(
    <DataContext.Provider value={providerProps}>
      {children}
      <div id="portal-root" />
    </DataContext.Provider>
  );
};
```

Worth reading up on inversion of control/dependency injection if you're not aware. ☝️This is how you would override values for the Contexts I've set up. `{providerProps}` can be `{data: myMockedData, save: myMockedFunction}`

so have a look at l203, I'm doing this:
```
const save = jest.fn();
   const { getByTestId, getByRole } = customRender(<LinkCreate />, {
     data,
     save, 
   });
```

`jest.fn` is doing the mocking here